### DISCUSSION TITLE
user or project not found

### DISCUSSION THREAD

**Discussion by matiux** [badges/shields#6192] [Q&A]
I'm trying to use shields.io on a private azure devops repo.
I have setup shields locally via docker(using the following guideline: https://github.com/badges/shields/blob/master/doc/self-hosting.md#docker) and I have set the AZURE_DEVOPS_TOKEN environment variabile with my PAT.

 
Unfortunately, when I try to generate the repo's build badge, a badge with the following error message is generated: "user or project not found"

 
This is the shield's url I'm using to generate the repo's build badge:
http://localhost:8080/azure-devops/build/[organization]/[project id]/[definition id]

**Comment by calebcartwright** [comment]
thanks for reaching out! it's incredibly difficult for us to be able to do much in the way of debugging/troubleshooting private self-hosted instances, and all I can tell you definitively is that badge message is indicative of the Azure DevOps API returning a 404 response.

You may want to try hitting the upstream endpoint directly (either in your web browser or with a client like curl) with the exact same values you are providing for the badge route parameters:

https://dev.azure.com/${organization}/${projectId}/_apis/build/status/${definitionId}

e.g. 

https://dev.azure.com/swellaby/opensource/_apis/build/status/25

**Reply by matiux** [reply]
Thanks for your reply @calebcartwright .

I have try to call the endpoint that you have suggested to me.
Let's say for example these I have this parameters:
* organization: foo 
* project name: Bar (with capital letter) 
* project id: 8cf3ec0e-d0c2-4fcd-8206-ad204f254a96
* definition id: 12345

So I tried calling:

With Project name:
**https://dev.azure.com/foo/Bar/_apis/build/status/12345**
```
I get the badge: Azure Pipelines: never built
```

With Project id:
**https://dev.azure.com/foo/8cf3ec0e-d0c2-4fcd-8206-ad204f254a96/_apis/build/status/12345**
```
I get the badge: Azure Pipelines: never built
```

With wrong Project name:
**https://dev.azure.com/foo/wrong_project_name/_apis/build/status/12345**
```
{"count":1,"value":{"Message":"Build pipeline 1235 was not found."}}
```

With wrong Project id:
**https://dev.azure.com/foo/wrong_project_id/_apis/build/status/12345**

```
{"$id":"1","innerException":null,"message":"VS800075: The project with id '8cf3ec0e-d0c2-4fcd-8206-ad204f254a96' does not exist, or you do not have permission to access it.","typeName":"Microsoft.TeamFoundation.Core.WebApi.ProjectDoesNotExistException, Microsoft.TeamFoundation.Core.WebApi","typeKey":"ProjectDoesNotExistException","errorCode":0,"eventId":3000}
```


Obviously I'm doing this calls in a browser with authenticated session on Azure dev ops. If I try to call these endpoints without authentication I get this error:
```
{"count":1,"value":{"Message":"Build pipeline 1235 was not found."}}
```

That's the most similar error that I get from shild.io: "user or project not found"


It seems that the calls that I'm doing from shilds.io are not authenticated. 

But the variable with PAT I set it. If I did it right ...

**Reply by calebcartwright** [reply]
Is the response status code for that last call (sans auth) a 404? What happens when you make the same call but with the auth header using the same PAT you've configured in your self-hosted Shields server?
